walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -7,17 +7,105 @@
 //     https://www.apache.org/licenses/LICENSE-2.0
 //
 // Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
+// distimeRangesAvailibuted under the License is distimeRangesAvailibuted on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
 package com.google.sps;
 
+import java.util.Queue;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.Comparator;
+import java.util.ArrayList;
+import java.util.List;
+
+/* A meeting request has:
+   - a name
+   - a duration in minutes
+   - a collection of attendees
+   
+   For a particular time slot to work, all attendees must be free to attend the meeting. When a query is made, it will be given a collection of all known events. Each event has:
+   - a name
+   - a time range
+   - a collection of attendees
+   A time range will give you the start time, the end time, and the duration of the event. If you want to know more, open the TimeRange.java file.
+*/
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    // Creates an ArrayList to hold busy times
+    List<TimeRange> busyTimes = new ArrayList<>();
+
+    // Gets info of request
+    Collection<String> attendees = request.getAttendees();
+    long dur = request.getDuration();
+
+    // Special case where there can be no meetings of that duration 
+    if(dur > TimeRange.END_OF_DAY - TimeRange.START_OF_DAY || dur < 0){
+      return busyTimes; 
```

Comment:
This is nitpicky, but I would probably return a new `ArrayList` here rather than returning `busyTimes`. My thinking is that `busyTimes` is used to hold the unavailable times, and even though it's empty at this point, "semantically" it doesn't represent what you want to return.

To put it another way: let's say for some reason the company had a new rule: no meetings during lunch. To enforce that, somebody added 12:00 - 1:00 to `busyTimes` on line 42. Now if somebody requested an invalid duration, the function would return 12:00 - 1:00 as an available time, even though it's not.

That's a contrived example, and hopefully a test would catch it anyway. But the idea is that I would keep `busyTimes` and the return value separate, even when they'll happen to be the same thing.